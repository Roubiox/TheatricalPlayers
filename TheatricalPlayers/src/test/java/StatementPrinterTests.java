import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;

import static org.approvaltests.Approvals.verify;

public class StatementPrinterTests {

    @Test
    void exampleStatement() {

        List<Play> plays = new ArrayList<>();
        plays.add(new Tragedie("Hamlet"));
        plays.add(new Comedie("As You Like It"));
        plays.add(new Tragedie("Othello"));

        ArrayList<Performance> temp = new ArrayList<>();
        temp.add(new Performance(plays.get(0), 55));
        temp.add(new Performance(plays.get(1), 35));
        temp.add(new Performance(plays.get(2), 40));

        Acheteur acheteur = new Acheteur("BigCo");
        Facture facture = new Facture(acheteur, temp);

        var result = facture.printToText();
        facture.printToHTML();
        verify(result);

    }
}