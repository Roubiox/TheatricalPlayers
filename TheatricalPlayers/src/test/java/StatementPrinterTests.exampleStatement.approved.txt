Declaration pour BigCo
  Hamlet: $650.00 (55 seats)

  As You Like It: $580.00 (35 seats)

  Othello: $500.00 (40 seats)

Le montant du est $1,730.00
Vous avez gagne 47 credits